prow/gerrit/adapter/adapter.go

```diff
@@ -425,18 +426,43 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		labels      map[string]string
 		annotations map[string]string
 	}
-
 	var jobSpecs []jobSpec
-
-	changedFiles := client.ChangedFilesProvider(&change)
+	baseSHAGetter := func() (string, error) { return baseSHA, nil }
+	var hasRelatedChanges *bool
+	// This headSHAGetter will return the empty string instead of the head SHA in cases where we can be certain that change does not
+	// modify inrepoconfig. This allows multiple changes to share a ProwYAML cache entry so long as they don't touch inrepo config themselves.
+	headSHAGetter := func() (string, error) {
+		changes, err := client.ChangedFilesProvider(&change)()
+		if err != nil {
+			return "", err
+		}
+		if config.ContainsInRepoConfigPath(changes) {
+			return change.CurrentRevision, nil
+		}
+		if hasRelatedChanges == nil {
+			if res, err := c.gc.HasRelatedChanges(instance, change.ChangeID, change.CurrentRevision); err != nil {
+				return "", err
```

Comment:
this is best effort in my opinion, it should be safe to fallback to CurrentRevision